Source/Core/VideoCommon/TextureConverterShaderGen.cpp

```diff
@@ -91,12 +92,21 @@ ShaderCode GenerateShader(APIType api_type, const UidData* uid_data)
 
   // The copy filter applies to both color and depth copies. This has been verified on hardware.
   // The filter is only applied to the RGB channels, the alpha channel is left intact.
-  out.Write("  float4 prev_row = SampleEFB(uv0, -1.0f);\n"
-            "  float4 current_row = SampleEFB(uv0, 0.0f);\n"
-            "  float4 next_row = SampleEFB(uv0, 1.0f);\n"
-            "  float4 texcol = float4(prev_row.rgb * filter_coefficients[0] +\n"
-            "                         current_row.rgb * filter_coefficients[1] +\n"
-            "                         next_row.rgb * filter_coefficients[2], current_row.a);\n");
+  if (uid_data->copy_filter)
+  {
+    out.Write("  float4 prev_row = SampleEFB(uv0, -1.0f);\n"
+              "  float4 current_row = SampleEFB(uv0, 0.0f);\n"
+              "  float4 next_row = SampleEFB(uv0, 1.0f);\n"
+              "  float4 texcol = float4(prev_row.rgb * filter_coefficients[0] +\n"
```

Comment:
As uniform branching is very fast, we might also try the lazy way:
if (filter_coefficients[0]) prev_row = SampleEFB(uv0, -1.0f);

Reply:
This ensures the texture sample does not execute at all, right? I was considering this too, but wasn't sure if it was guaranteed to be skipped.

Reply:
No, it doesn't guarantee it. The driver is still allowed to slow down, but I see no reason why it should do so.